### PR TITLE
Correct Account model for rubocop

### PR DESCRIPTION
**Pull request by tsuwatch**
before
```sh
Offenses:

app/models/account.rb:3:1: C: Class has too many lines. [285/200]
class Account < ApplicationRecord ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:66:3: W: Parenthesize the param -> { where.not(domain: nil) } to make sure that the block will be associated with the -> method call.
  scope :remote, -> { where.not(domain: nil) }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:67:3: W: Parenthesize the param -> { where(domain: nil) } to make sure that the block will be associated with the -> method call.
  scope :local, -> { where(domain: nil) }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:68:3: W: Parenthesize the param -> { where('(select count(f.id) from follows as f where f.target_account_id = accounts.id) = 0') } to make sure that the block will be associated with the -> method call.
  scope :without_followers, -> { where('(select count(f.id) from follows as f where f.target_account_id = accounts.id) = 0') }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:69:3: W: Parenthesize the param -> { where('(select count(f.id) from follows as f where f.target_account_id = accounts.id) > 0') } to make sure that the block will be associated with the -> method call.
  scope :with_followers, -> { where('(select count(f.id) from follows as f where f.target_account_id = accounts.id) > 0') }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:70:3: W: Parenthesize the param ->(time) { where(subscription_expires_at: nil).or(where('subscription_expires_at < ?', time)).remote.with_followers } to make sure that the block will be associated with the -> method call.
  scope :expiring, ->(time) { where(subscription_expires_at: nil).or(where('subscription_expires_at < ?', time)).remote.with_followers }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:71:3: W: Parenthesize the param -> { where(silenced: true) } to make sure that the block will be associated with the -> method call.
  scope :silenced, -> { where(silenced: true) }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:72:3: W: Parenthesize the param -> { where(suspended: true) } to make sure that the block will be associated with the -> method call.
  scope :suspended, -> { where(suspended: true) }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:73:3: W: Parenthesize the param -> { reorder(id: :desc) } to make sure that the block will be associated with the -> method call.
  scope :recent, -> { reorder(id: :desc) }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:74:3: W: Parenthesize the param -> { order(domain: :asc, username: :asc) } to make sure that the block will be associated with the -> method call.
  scope :alphabetic, -> { order(domain: :asc, username: :asc) }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:75:3: W: Parenthesize the param -> { group(:domain).select(:domain, 'COUNT(*) AS accounts_count').order('accounts_count desc') } to make sure that the block will be associated with the -> method call.
  scope :by_domain_accounts, -> { group(:domain).select(:domain, 'COUNT(*) AS accounts_count').order('accounts_count desc') }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:78:5: C: Align the parameters of a method call if they span more than one line.
    :current_sign_in_ip,
    ^^^^^^^^^^^^^^^^^^^
app/models/account.rb:79:5: C: Align the parameters of a method call if they span more than one line.
    :current_sign_in_at,
    ^^^^^^^^^^^^^^^^^^^
app/models/account.rb:80:5: C: Align the parameters of a method call if they span more than one line.
    :confirmed?,
    ^^^^^^^^^^^
app/models/account.rb:81:5: C: Align the parameters of a method call if they span more than one line.
    to: :user, ...
    ^^^^^^^^^^
app/models/account.rb:145:5: C: Use subscription_expires_at.present? instead of !subscription_expires_at.blank?.
    !subscription_expires_at.blank?
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:197:16: C: %w-literals should be delimited by [ and ].
    return if !%w(http https).include?(parsed_url.scheme) || parsed_url.host.empty? || self[:avatar_remote_url] == url
               ^^^^^^^^^^^^^^
app/models/account.rb:208:16: C: %w-literals should be delimited by [ and ].
    return if !%w(http https).include?(parsed_url.scheme) || parsed_url.host.empty? || self[:header_remote_url] == url
               ^^^^^^^^^^^^^^
app/models/account.rb:332:26: C: Prefer reduce over inject.
      query.pluck(field).inject({}) { |mapping, id| mapping[id] = true; mapping }
                         ^^^^^^
app/models/account.rb:332:26: C: Use each_with_object instead of inject.
      query.pluck(field).inject({}) { |mapping, id| mapping[id] = true; mapping }
                         ^^^^^^
app/models/account.rb:332:71: C: Do not use semicolons to terminate expressions.
      query.pluck(field).inject({}) { |mapping, id| mapping[id] = true; mapping }
                                                                      ^
app/models/account.rb:368:5: C: Use if avatar.present? instead of unless avatar.blank?.
    unless avatar.blank?
    ^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:374:5: C: Use if header.present? instead of unless header.blank?.
    unless header.blank?
    ^^^^^^^^^^^^^^^^^^^^
app/models/account.rb:374:5: C: Use a guard clause instead of wrapping the code inside a conditional expression.
    unless header.blank?
    ^^^^^^

1 file inspected, 24 offenses detected
```
after
```sh
Offenses:

app/models/account.rb:3:1: C: Class has too many lines. [284/200]
class Account < ApplicationRecord ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -63,24 +63,24 @@ class Account < ApplicationRecord
...
-  scope :local, -> { where(domain: nil) }
-  scope :without_followers, -> { where('(select count(f.id) from follows as f where f.target_account_id = accounts.id) = 0') }
-  scope :with_followers, -> { where('(select count(f.id) from follows as f where f.target_account_id = accounts.id) > 0') }
-  scope :expiring, ->(time) { where(subscription_expires_at: nil).or(where('subscription_expires_at < ?', time)).remote.with_followers }
-  scope :silenced, -> { where(silenced: true) }
-  scope :suspended, -> { where(suspended: true) }
-  scope :recent, -> { reorder(id: :desc) }
-  scope :alphabetic, -> { order(domain: :asc, username: :asc) }
-  scope :by_domain_accounts, -> { group(:domain).select(:domain, 'COUNT(*) AS accounts_count').order('accounts_count desc') }
+  scope :remote, (-> { where.not(domain: nil) })
```
Comment (mjankowski) [inline app/models/account.rb]: I think I'd probably prefer that we just remove this rule from rubocop styles...

File: app/models/account.rb [hunk 2]
```diff
@@ -63,24 +63,24 @@ class Account < ApplicationRecord
...
+  scope :by_domain_accounts, (-> { group(:domain).select(:domain, 'COUNT(*) AS accounts_count').order('accounts_count desc') })
 
   delegate :email,
-    :current_sign_in_ip,
-    :current_sign_in_at,
-    :confirmed?,
-    to: :user,
-    prefix: true,
-    allow_nil: true
+           :current_sign_in_ip,
```
Comment (mjankowski) [inline app/models/account.rb]: I'm generally skeptical of commits/changes that JUST fix rubocop warnings w/out also improving readability ... I'd put these indentation ones in that category.

File: spec/models/account_spec.rb [hunk 3]
```diff
@@ -326,8 +326,12 @@
   end
 
   describe '.following_map' do
+    let(:account) { Fabricate(:account) }
+    let(:target) { Fabricate(:account) }
+    let!(:follow) { Fabricate(:follow, account: account, target_account: target) }
+
     it 'returns an hash' do
-      expect(Account.following_map([], 1)).to be_a Hash
+      expect(Account.following_map([target.id], account.id)).to eq({ "#{target.id}".to_i => true })
```
Comment (mjankowski) [inline spec/models/account_spec.rb]: Can you explain the `"#{target.id}".to_i` thing ... shouldn't `target.id` already be an integer?

Also - it looks like this change is adding spec coverage (thanks!) and is not just a style fix?  Maybe put it in a separate commit or separate PR?

**Review by mjankowski — DISMISSED** [review]
Left a few remarks on things to change.

File: app/models/account.rb (hunk 1 above)
Comment (tsuwatch) [inline app/models/account.rb]: Okay, remove these changes

File: spec/models/account_spec.rb (hunk 3 above)
Comment (tsuwatch) [inline spec/models/account_spec.rb]: It was redundant.
Okay, remove this change in this PR.

File: app/models/account.rb (hunk 2 above)
Comment (tsuwatch) [inline app/models/account.rb]: I see

**Comment by tsuwatch** [comment]
rebased, https://github.com/tootsuite/mastodon/pull/2630 was merged earlier... :open_mouth:

**Review by Gargron — APPROVED** [review]
